### DISCUSSION TITLE
How to check for errors with Extend JS?

### DISCUSSION THREAD

**Discussion by TheHypnoo** [pocketbase/pocketbase#5692] [Q&A]
Hi, I would like to know if there is a way to ‘control’ the errors.

With the expand pocketbase function in Javascript. I want to try to see if an element exists, and if it doesn't exist I want to create it.
Is the solution to try to get the element inside a try catch? 

for example:
try {
    const sensor = $app.dao().findFirstRecordByData(‘sensors’, ‘id’, c.pathParam(‘id’))
} catch (error) {
console.log(error)
}

What happens, is that it doesn't return the console.log(error) any different than removing the try catch... 

What would be the solution?

Thank you!

**Comment by loopByte** [comment]
Hello! I have the same problem right now. I want to use the hooks to create/update records on another collection after I update a record on another collection. I am using `onRecordAfterUpdateRequest` hook and I need to check if some entries exists in the database and update them, if not, to create them. Did you find an asnwer regarding this?

**Comment by chnfyi** [comment]
Based on my understanding, one solution could be :

```javascript
onRecordAfterUpdateRequest((e) => {
   // Update on "users" collection
   
   // Retrieve value from a specific field
   const data = e.record.get("someField")
   let record
   
   try {
   // Every single record retrieval method throws an error if no item found
      record = $app.dao().findRecordById("otherCollection", "id")
   } catch (error) {
      // If no item has been found, create it
      const collection = $app.dao().findCollectionByNameOrId("otherCollection")
      record = new Record(collection, {
         "title": "Lorem ipsum",
         "fieldToUpdate": ""
      })
   }

   // Update your existing/new "otherCollection" record and save it
   record.set("fieldToUpdate", data)
   $app.dao().saveRecord(record)
}, "users")
```

**Reply by loopByte** [reply]
Hi! Thank you for your response. I can confirm that this solution worked for me. Thank you!

**Reply by TheHypnoo** [reply]
Yes, that may be a solution. But the problem with that is that you don't really know why the error is there. The only way I could solve it is:

    const recordNotFound = 'no rows in result set'

try {
...
} catch (error) {
 if (error.message.includes(recordNotFound)) {
        createSensor()
        runFlowAfterSensorUpdate()
        return c.json(200, { message: 'Sensor created' })
      } else {
        console.log('ERROR', error)
        return c.json(500, { error: error.message })
      }
}

Since it is specified in the question that there was no result... But I don't think it's quite right to check it by message, but I didn't find any other way after 1 week of research.

**Reply by loopByte** [reply]
Yes, you don't have a very good idea why your request failed, but since you just look in the table/collection for a row/record, the result would be false/null/undefined and therefore the hook fails and you don't need more reasons or checking since there are no records i n the collection.
